Fixed: Order history pagination (#4599)

AishwaryShrivastav opened the pull request:
### Related Issues
<!--  Put related issue number which this PR is closing. For example #123 -->

closes #4599 

### Short Description and Why It's Useful
<!-- describe in a few words what is this Pull Request changing and why it's useful -->

the getOrdersHistory, which previously used to fetch all the orders, after introducing pagination support in "getOrdersHistory", it only fetches 20 orders, which is its default value.

Still, the order history page calls it only once, trying to fetch all the results, and on those results, it applies lazyLoadOrdersOnScroll.

Due to which the user can not see more than 20 orders.

### Screenshots of Visual Changes before/after (if There Are Any)
<!-- if you made any changes in the UI layer please provide before/after screenshots -->

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature
- [ ] (Next only) I've followed this [instruction](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and I've created a `.js` file with information about my Pull Request

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance
<!-- Please get familiar with following info -->

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vuestorefront/vue-storefront?pullRequest=5527) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vuestorefront/vue-storefront?pullRequest=5527) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vuestorefront/vue-storefront?pullRequest=5527) it.</sub>

AishwaryShrivastav commented:
Created a new action for user state to append orders in the state, which will only get calls from pagination and remove the previously added appending logic from refreshOrderHistory.
We had added appending logic for pagination on getOrderHistory, it fetches order history and appends to the state, if state already had any orders, but it created these issues:

1. On every window refresh, getOrderHistory runs and appends the same orders again into the state. 

2. On placing the order,  getOrderHistory runs and appends the new order again into the state.  


So, For pagination, we wrote another action only append orders and will only run from pagination and we removed the changes from getOrderHistory ( we added to append the state, in case of older orders in state) so that older action will always refresh the state as we wrote a new action which will only append the state and only get call from pagination.

lukaszjedrasik commented on core/modules/order/components/UserOrdersHistory.ts:
```diff
@@ -26,6 +27,14 @@ export default {
     onBottomScroll () {
       ++this.pagination.current
-    }
+      const totalCount = this.$store.state.user.orders_history.totalCount ? this.$store.state.user.orders_history.totalCount : 0
```
Attribute name is `total_count`, not `totalCount`
```js
this.$store.state.user.orders_history.total_count ? this.$store.state.user.orders_history.total_count : 0
```

lukaszjedrasik commented on core/modules/order/components/UserOrdersHistory.ts:
```diff
@@ -26,6 +27,14 @@ export default {
+      const totalCount = this.$store.state.user.orders_history.totalCount ? this.$store.state.user.orders_history.totalCount : 0
+      const isLastPage = this.pagination.current > Math.ceil(totalCount / this.pagination.perPage);
+      if (!isLastPage) {
+        this.appendNewOrders({ currentPage: this.pagination.current, refresh: true })
```
1. I think that `refresh: true` is unnecessary.  
2. Argument type `{ refresh: boolean, currentPage: any }` is not assignable to parameter type `{ resolvedFromCache: any, pageSize?: any, currentPage?: any }` 

![Zrzut ekranu 2021-04-22 o 12 07 38](https://user-images.githubusercontent.com/40273258/115696863-58182800-a363-11eb-84da-64f187d92d43.png)

lukaszjedrasik commented on core/modules/user/store/actions.ts:
```diff
@@ -224,6 +224,7 @@ const actions: ActionTree<UserState, RootState> = {
     commit(types.USER_GROUP_TOKEN_CHANGED, '')
     commit(types.USER_GROUP_CHANGED, null)
     commit(types.USER_INFO_LOADED, null)
+    commit(types.USER_ORDERS_HISTORY_LOADED, null)
```
It is duplicated. Look at line `233`.

lukaszjedrasik reviewed (changes requested):
I think that problem still occurs: https://github.com/vuestorefront/vue-storefront/pull/4632#issuecomment-692658418

So please do a lot of manual tests.

lukaszjedrasik commented:
New PR: https://github.com/vuestorefront/vue-storefront/pull/5850
